Bottom: 7a63e115cc4adf073292cd3edca568d9910e1a6e
Top:    145c31a6954c193a2ed1d3716d0fb4db0b11cd14
Author: Dotan Barak <dotanb@dev.mellanox.co.il>
Date:   2012-10-09 12:27:48 +0000

[1/5,librdmacm] rsocket: add missing va_end() after calling va_end()

Not doing so, may lead to resource leak.

Signed-off-by: Dotan Barak <dotanb@dev.mellanox.co.il>
Reviewed-by: Yishai Hadas <yishaih@mellanox.com>
Signed-off-by: Sean Hefty <sean.hefty@intel.com>


---

diff --git a/src/rsocket.c b/src/rsocket.c
index 4c4821b..5b65b12 100644
--- a/src/rsocket.c
+++ b/src/rsocket.c
@@ -2002,7 +2002,8 @@ int rfcntl(int socket, int cmd, ... /* arg */ )
 	va_start(args, cmd);
 	switch (cmd) {
 	case F_GETFL:
-		return (int) rs->fd_flags;
+		ret = (int) rs->fd_flags;
+		break;
 	case F_SETFL:
 		param = va_arg(args, long);
 		if (param & O_NONBLOCK)
@@ -2013,6 +2014,7 @@ int rfcntl(int socket, int cmd, ... /* arg */ )
 		break;
 	default:
 		ret = ERR(ENOTSUP);
+		break;
 	}
 	va_end(args);
 	return ret;
